components/mailchimp/sources/new-open/new-open.js

```diff
@@ -0,0 +1,97 @@
+const common = require("../common/timer-based");
+
+module.exports = {
+  ...common,
+  key: "new-open",
+  name: "New Open",
+  description:
+    "Emit new event when a recipient opens an email in a specific campaign.",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    campaignId: {
+      propDefinition: [
+        common.props.mailchimp,
+        "campaignId",
+      ],
+      description:
+      "The unique ID of the campaign you'd like to watch for new email opens on links.",
+    },
+    uniqueOpensOnly: {
+      type: "boolean",
+      label: "Unique Opens Only?",
+      description: "Whether to count every open campaign email or only count opens coming from each subscriber only once",
+      default: false,
+    },
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      return await this.emitReportSampleEvents(this.campaignId, undefined, (new Date()).getTime());
+    },
+  },
+  methods: {
+    ...common.methods,
+    getEventTypes() {
+        return [
+          "opens",
+        ];
+      },
+    generateMeta({
+      eventPayload,
+      diff: openDiff,
+      timestamp: ts,
+    }) {
+      const { campaign_id: campaignId } = eventPayload;
+      return {
+        id: `${campaignId}-${ts}`,
+        summary: `${openDiff} new opens`,
+        ts,
+      };
+    },
+    getCachedCampaignDetails(){
+      return {
+        total_opens: this.getDbServiceVariable("total_opens"),
+        members: this.getDbServiceVariable("members"),
+      };
+    },
+    async getCampaignDetails(){
+      return await this.mailchimp.listCampaignOpenDetails(
+        this.campaignId,
+      );
```

Comment:
This `await` can be removed
```suggestion
    async getCampaignDetails(){
      return this.mailchimp.listCampaignOpenDetails(
        this.campaignId,
      );
```